File: tech-studio-projects/financial-industry/algo-trading-platform/java/algo-trading-platform/src/com/fintech/algotrading/logging/TextLogger.java
package com.fintech.algotrading.logging;

public class TextLogger {

    private String directory;
    private String filename;
    private String fileExtension;

    public String getDirectory() {
        return directory;
    }

    public void setDirectory(String dir) {
        this.directory = dir;
    }

    public String getFilename() {
        return filename;
    }

    public void setFilename(String file) {
        this.filename = file;
    }

    public String getFileExtension() {
        return fileExtension;
    }

    public void setFileExtension(String ext) {
        this.fileExtension = ext;
    }
}

File: tech-studio-projects/financial-industry/algo-trading-platform/cpp/logging/TextLogger.cpp
#include "TextLoggerConfiguration.h"

// Getter for Directory
const std::string& TextLoggerConfiguration::GetDirectory() const {
    return directory;
}

// Setter for Directory
void TextLoggerConfiguration::SetDirectory(const std::string& dir) {
    directory = dir;
}

// Getter for Filename
const std::string& TextLoggerConfiguration::GetFilename() const {
    return filename;
}

// Setter for Filename
void TextLoggerConfiguration::SetFilename(const std::string& file) {
    filename = file;
}

// Getter for FileExtension
const std::string& TextLoggerConfiguration::GetFileExtension() const {
    return fileExtension;
}

// Setter for FileExtension
void TextLoggerConfiguration::SetFileExtension(const std::string& ext) {
    fileExtension = ext;
}
